<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="agent_1"  default="tb3_1"/>
  <arg name="agent_2" default="tb3_2"/>

  <arg name="agent_1_x_pos" default=" 0.0"/>
  <arg name="agent_1_y_pos" default="-10.0"/>
  <arg name="agent_1_z_pos" default=" 0.0"/>
  <arg name="agent_1_yaw"   default=" 1.57"/>

  <arg name="agent_2_x_pos" default=" 9.0"/>
  <arg name="agent_2_y_pos" default=" 0.0"/>
  <arg name="agent_2_z_pos" default=" 0.0"/>
  <arg name="agent_2_yaw"   default=" 3.14"/>


  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find collision_predictor)/world2.sdf"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>  

  <group ns = "$(arg agent_1)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg agent_1)" />
    </node>
    
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg agent_1) -x $(arg agent_1_x_pos) -y $(arg agent_1_y_pos) -z $(arg agent_1_z_pos) -Y $(arg agent_1_yaw) -param robot_description" />
  </group>

  <group ns = "$(arg agent_2)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg agent_2)" />
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg agent_2) -x $(arg agent_2_x_pos) -y $(arg agent_2_y_pos) -z $(arg agent_2_z_pos) -Y $(arg agent_2_yaw) -param robot_description" />
  </group>

</launch>
